Interface 'ErrorEvent' incorrectly extends interface 'Event'

### Is there an existing issue for this?

- [X] I have checked for existing issues https://github.com/getsentry/sentry-javascript/issues
- [X] I have reviewed the documentation https://docs.sentry.io/
- [X] I am using the latest SDK release https://github.com/getsentry/sentry-javascript/releases

### How do you use Sentry?

Sentry Saas (sentry.io)

### Which package are you using?

@sentry/react

### SDK Version

7.27.0

### Framework Version

react 18.2.0

### Link to Sentry event

_No response_

### Steps to Reproduce

Compiling some React code with a Sentry dependency using `tsc` with Typescript 4.9.4 results in this error message:
```
../node_modules/.pnpm/@sentry+types@7.27.0/node_modules/@sentry/types/types/event.d.ts:65:18 - error TS2430: Interface 'ErrorEvent' incorrectly extends interface 'Event'.
  Types of property 'type' are incompatible.
    Type 'undefined' is not assignable to type 'EventType'.

65 export interface ErrorEvent extends Event {
                    ~~~~~~~~~~


Found 1 error in ../node_modules/.pnpm/@sentry+types@7.27.0/node_modules/@sentry/types/types/event.d.ts:65
```

A possible fix is to replace `type?: EventType;` by `type: EventType | undefined;` on line 47 of `event.d.ts`

### Expected Result

No error


### Actual Result

tsc compilation error

--- a/packages/types/src/event.ts
+++ b/packages/types/src/event.ts
@@ -58,8 +58,12 @@ export interface Event {
   };
 }
 
-/** JSDoc */
-export type EventType = 'transaction' | 'profile';
+/**
+ * The type of an `Event`.
+ * Note that `ErrorEvent`s do not have a type (hence its undefined),
+ * while all other events are required to have one.
+ */
+export type EventType = 'transaction' | 'profile' | undefined;
 
 export interface ErrorEvent extends Event {
   type: undefined;

